docs/source/tutorials/open_images_evaluation.ipynb

```diff
@@ -0,0 +1,444 @@
+{
+ "cells": [
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "# Object Detection Model Evaluation on Google's Open Images Dataset\n",
+    "\n",
+    "This tutorial demonstrates per-image evaluation of an object detection model on [the Open Images dataset](https://storage.googleapis.com/openimages/web/index.html)\n",
+    "that generates:\n",
+    "\n",
+    "- true positives & false positives\n",
+    "- per-class average precision (AP)\n",
+    "- mean average precision (mAP)\n",
+    "\n",
+    "for each image and adds this information to each [Sample](https://voxel51.com/docs/fiftyone/api/fiftyone.core.sample.html#fiftyone.core.sample.Sample)\n",
+    "in the [Dataset](https://voxel51.com/docs/fiftyone/api/fiftyone.core.dataset.html#fiftyone.core.dataset.Dataset).\n",
+    "\n",
+    "The steps are broken down as follows:\n",
+    "\n",
+    "1. [Requirements](#Requirements)\n",
+    "2. [Download the test data and ground-truth labels](#Download-the-test-data-and-ground-truth-labels) (optional)\n",
+    "3. [Generate predictions](#Generate-predictions) (optional)\n",
+    "4. [Load the data into FiftyOne](#Load-the-data-into-FiftyOne)\n",
+    "5. [Prepare the ground-truth for evaluation](#Prepare-the-ground-truth-for-evaluation) (optional)\n",
+    "6. [Evaluate on a per-image granularity](#Evaluate-on-a-per-image-granularity)\n",
+    "7. [Explore](#Explore)\n",
+    "\n",
+    "Optional steps may not be necessary depending on if you have already downloaded the data or have your own model to evaluate.\n",
+    "\n",
+    "This tutorial evaluates a model on [Open Images V4](https://storage.googleapis.com/openimages/web/download_v4.html)\n",
+    "however this code supports later versions of Open Images as well. If using a newer version just make sure to\n",
+    "use the appropriate hierarchy file and class label map."
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "## Requirements\n",
+    "\n",
+    "This notebook contains bash commands. To run it as a notebook, you must install the [Jupyter bash kernel](https://github.com/takluyver/bash_kernel) via the command below.\n",
+    "\n",
+    "Alternatively, you can just copy + paste the code blocks into your shell."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 1,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "pip install bash_kernel\n",
+    "python -m bash_kernel.install"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "This workflow requires a few required python packages.\n",
+    "\n",
+    "Install the appropriate version of `tensorflow` depending on whether or not you\n",
+    "have a GPU:"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "pip install tensorflow\n",
```

Comment:
So many tensorflow issues..... I started with v1.12.0 and it gave me a `ModuleNotFoundError: No module named 'tensorflow.python.training.tracking'` when trying to run inference. 

v1.13 gave me cuda issues

v2.1 finally worked for me but then I ran into other issues with `tensorflow_hub` where it was trying to access `/tmp` and couldn't.

It seems like the only outcome of this is to generate `predictions.csv`. I think you should just provide that to be able to run this tutorial quickly and then keep the the inference only for people who really want to run their own model.

Honestly, it seems like you could remove any examples of running inference or using `tensorflow_hub` from the tutorial and just mention that you have a script available if someone would want to do that. I think provding a `predictions.csv` file and describing the format of it is enough.

Reply:
I changed this to `pip install --upgrade tensorflow` which should resolve version issues. General comment: for the most part in 2020, if you try to use tensorflow v1.X you're going to have a bad time.

What was the issue with `/tmp`?

I also changed the title of the section to `(optional) Generate predictions` and made "**Alternatively**, ..." bold so that it's a little more attention grabbing.

But I think it's fine to leave this section and not have a `predictions.csv` download. The entire purpose of this tutorial is to be able to do this all yourself, in particular, to evaluate your own model. I would like to get the download with 500 images with predictions to be available and probably link that in the top of this tutorial though

Reply:
I still feel pretty adamant about providing a `predictions.csv` file.

1) Sure, some people go through tutorials to be able to replicate exactly what is in them for their purposes, but there is also a good portion of people that go through tutorials to just learn about the tool. Having precomputed predictions makes using FiftyOne in this tutorial way easier. For example, if someone just wants to poke around Open Images. 

(^this is addressed by providing a zip)

2) It doesn't let you use your model directly anyway. The provided code lets you generate predictions for any `tensorflow_hub` model, but most people don't use that and would have to generate predictions off of the `csv` file structure you presented. It would make it a lot easier to see how the rest of the code works with example predictions to let people know if they want to go through the trouble of generating their own.

(^not addressed by zip)

3) Cutting down on bash code that needs to be run would make it more feasible to turn this into a python notebook where you copy the bash blocks. This would let the notebook require less setup.

(^not addressed by zip)

Reply:
if you trust me...let's get this merged as is, and once we've sorted out hosting I'll put a predictions file up on the website for download (and update this tutorial)